tobler/util/util.py

```diff
@@ -102,14 +106,76 @@ def project_gdf(gdf, to_crs=None, to_latlong=False):
 
     # calculate the centroid of the union of all the geometries in the
     # GeoDataFrame
-    avg_longitude = gdf['geometry'].unary_union.centroid.x
+    avg_longitude = gdf["geometry"].unary_union.centroid.x
 
     # calculate the UTM zone from this avg longitude and define the UTM
     # CRS to project
-    utm_zone = int(math.floor((avg_longitude + 180) / 6.) + 1)
-    utm_crs = '+proj=utm +zone={} +ellps=WGS84 +datum=WGS84 +units=m +no_defs'.format(utm_zone)
+    utm_zone = int(math.floor((avg_longitude + 180) / 6.0) + 1)
+    utm_crs = "+proj=utm +zone={} +ellps=WGS84 +datum=WGS84 +units=m +no_defs".format(
+        utm_zone
+    )
 
     # project the GeoDataFrame to the UTM CRS
     projected_gdf = gdf.to_crs(utm_crs)
 
     return projected_gdf
+
+
+def hexify(source, resolution=6, clip=False):
+    """Generate a hexgrid geodataframe that covers the face of a source geodataframe.
+
+    Parameters
+    ----------
+    source : geopandas.GeoDataFrame
+        GeoDataFrame to transform into a hexagonal grid
+    resolution : int, optional
+        resolution of output h3 hexgrid. 
+        See <https://h3geo.org/docs/core-library/restable> for more information
+    clip : bool, optional
+        if True, hexagons are clipped to the precise boundary of the source gdf. Otherwise,
+        heaxgons along the boundary will be left intact.
+
```

Comment:
I'd add an option for `return_geoms` or similar that controls whether the actual polygons are returned. I'm not sure, we probably have to create them to do the apportioning but given that tools like Kepler.gl can ingest H3 ids and render them and that some folks might use this for fusing data, if we can find ways of apportioning without creating the geometries, we might be able to be more efficient, and not have to create/return a bunch of hex polygons that'll be discarded as soon as the table is created. Food for thought.